packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/EnumTypeProvider.cs

```diff
@@ -0,0 +1,363 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using static Microsoft.Generator.CSharp.Expressions.Snippets;
+
+namespace Microsoft.Generator.CSharp
+{
+    public class EnumTypeProvider : TypeProvider
+    {
+        private readonly IReadOnlyList<InputEnumTypeValue> _allowedValues;
+        private readonly ModelTypeMapping? _typeMapping;
+
+        public EnumTypeProvider(InputEnumType input, SourceInputModel? sourceInputModel) : base(sourceInputModel)
+        {
+            _allowedValues = input.AllowedValues;
+            _deprecated = input.Deprecated;
+
+            if (input.Accessibility == "internal")
+            {
+                DeclarationModifiers = TypeSignatureModifiers.Internal;
+            }
+            else
+            {
+                DeclarationModifiers = TypeSignatureModifiers.Public;
+            }
+
+            var isExtensible = input.IsExtensible;
+            if (ExistingType != null)
+            {
+                isExtensible = ExistingType.TypeKind switch
+                {
+                    TypeKind.Enum => false,
+                    TypeKind.Struct => true,
+                    _ => throw new InvalidOperationException(
+                        $"{ExistingType.ToDisplayString()} cannot be mapped to enum," +
+                        $" expected enum or struct got {ExistingType.TypeKind}")
+                };
+
+                _typeMapping = sourceInputModel?.CreateForModel(ExistingType);
+            }
+
+            if (isExtensible)
+            {
+                // extensible enums are implemented by readonly structs
+                DeclarationModifiers |= TypeSignatureModifiers.Partial | TypeSignatureModifiers.ReadOnly;
+            }
+
+            Name = input.Name.ToCleanName();
+            Namespace = GetDefaultModelNamespace(CodeModelPlugin.Instance.Configuration.Namespace);
+            IsExtensible = isExtensible;
+            ValueType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(input.EnumValueType);
+            IsStringValueType = ValueType.Equals(typeof(string));
+            IsIntValueType = ValueType.Equals(typeof(int)) || ValueType.Equals(typeof(long));
+            IsFloatValueType = ValueType.Equals(typeof(float)) || ValueType.Equals(typeof(double));
+            IsNumericValueType = IsIntValueType || IsFloatValueType;
+
+            Description = input.Description;
+
+            _valueField = new FieldDeclaration(FieldModifiers.Private | FieldModifiers.ReadOnly, ValueType, "_value");
+
+            if (!IsExtensible)
+            {
+                SerializationProviders = CodeModelPlugin.Instance.GetSerializationTypeProviders(this);
+            }
+        }
+
+        private readonly FieldDeclaration _valueField;
+
+        public CSharpType ValueType { get; }
+        public bool IsExtensible { get; }
+        public bool IsIntValueType { get; }
```

Comment:
I made this public because I put the "serialization" for enum in another project just like we did for models.
I will change that back as you mentioned in other comments and this should be resolved.

Reply:
They are internal again.